code/modules/power/supermatter/supermatter.dm

```diff
@@ -1,438 +1,876 @@
 
+	soundloop = new(list(src), TRUE)
 
-/obj/machinery/power/supermatter_shard/proc/handle_admin_warnings()
-	if(disable_adminwarn)
-		return
-
-	// Generic checks, similar to checks done by supermatter monitor program.
-	aw_normal = status_adminwarn_check(SUPERMATTER_NORMAL, aw_normal, "INFO: Supermatter crystal has been energised.<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>(JMP)</a>.", FALSE)
-	aw_notify = status_adminwarn_check(SUPERMATTER_NOTIFY, aw_notify, "INFO: Supermatter crystal is approaching unsafe operating temperature.<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>(JMP)</a>.", FALSE)
-	aw_warning = status_adminwarn_check(SUPERMATTER_WARNING, aw_warning, "WARN: Supermatter crystal is taking integrity damage!<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>(JMP)</a>.", FALSE)
-	aw_danger = status_adminwarn_check(SUPERMATTER_DANGER, aw_danger, "WARN: Supermatter integrity is below 75%!<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>(JMP)</a>.", TRUE)
-	aw_emerg = status_adminwarn_check(SUPERMATTER_EMERGENCY, aw_emerg, "CRIT: Supermatter integrity is below 50%!<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>(JMP)</a>.", FALSE)
-	aw_delam = status_adminwarn_check(SUPERMATTER_DELAMINATING, aw_delam, "CRIT: Supermatter is delaminating!<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>(JMP)</a>.", TRUE)
-
-/obj/machinery/power/supermatter_shard/proc/status_adminwarn_check(min_status, current_state, message)
-	var/status = get_status()
-	if(status >= min_status)
-		if(!current_state)
-			log_and_message_admins(message)
-			SSdiscord.send2discord_simple_noadmins(message)
-		return TRUE
-	else
-		return FALSE
-
-
-/obj/machinery/power/supermatter_shard/Destroy()
+/obj/machinery/power/supermatter_crystal/Destroy()
 	investigate_log("has been destroyed.", "supermatter")
-	if(damage > emergency_point)
-		emergency_lighting(0)
-	QDEL_NULL(radio)
-	GLOB.poi_list.Remove(src)
 	SSair.atmos_machinery -= src
+	QDEL_NULL(radio)
+	GLOB.poi_list -= src
+	QDEL_NULL(countdown)
+	QDEL_NULL(soundloop)
 	return ..()
 
-/obj/machinery/power/supermatter_shard/proc/explode()
-	investigate_log("has exploded.", "supermatter")
-	explosion(get_turf(src), explosion_power, explosion_power * 1.2, explosion_power * 1.5, explosion_power * 2, 1, 1)
-	qdel(src)
-	return
-
-/obj/machinery/power/supermatter_shard/process_atmos()
-	var/turf/L = loc
-
-	if(isnull(L))		// We have a null turf...something is wrong, stop processing this entity.
-		return PROCESS_KILL
+/obj/machinery/power/supermatter_crystal/examine(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
 
-	if(!istype(L)) 	//We are in a crate or somewhere that isn't turf, if we return to turf resume processing but for now.
-		return  //Yeah just stop.
+	var/range = HALLUCINATION_RANGE(power)
+	for(var/mob/living/carbon/human/H in viewers(range, src))
+		if(H != user)
+			continue
```

Comment:
Is this cheaper than `if(orange(src, user, range))`? It seems like it wont be, but I aint 100% sure

Reply:
altered completely.